Chapter07/pretty_print_on_the_fly.cpp

```diff
@@ -4,12 +4,15 @@
 using namespace std;
 
 class format_guard {
-    decltype(cout.flags()) f {cout.flags()};
+    ostream& os;
+    ios::fmtflags f;
 
 public:
-    ~format_guard() { cout.flags(f); }
+    explicit format_guard(ostream& os = cout) : os(os), f(os.flags()) {}
```

Comment:
Now one var is kind of aliasing the other... can you name the constructor parameter `os` maybe something like `guarded_stream` or similar?